firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,544 @@
+
+        self.ini_file_path = self.marionette.execute_script("""
+            Cu.import('resource://gre/modules/Services.jsm');
+
+            var file = Services.dirsvc.get('GreD', Ci.nsIFile);
+            file.append('update-settings.ini');
+
+            return file.path;
+
+            """)
+
+    @property
+    def config(self):
+        config = ConfigParser.ConfigParser()
+        config.readfp(open(self.ini_file_path))
+        return config
+
+    @property
+    def config_file_path(self):
+        return self.ini_file_path
+
+    @property
+    def config_file_contents(self):
+        with open(self.ini_file_path) as f:
+            return f.read()
+
+    @property
+    def channels(self):
+        channels = self.config.get(self.INI_SECTION, self.INI_OPTION)
+        return channels.split(',')
+
+    def set_channels(self, channels):
+        channels = ','.join(channels)
+        new_config = self.config
+        new_config.set(self.INI_SECTION, self.INI_OPTION, channels)
+        with open(self.ini_file_path, 'wb') as f:
+            new_config.write(f)
+
+    def add_channels(self, channels):
+        new_channels = ','.join(self.channels + channels)
+        new_config = self.config
+        new_config.set(self.INI_SECTION, self.INI_OPTION, new_channels)
+        with open(self.ini_file_path, 'wb') as f:
+            new_config.write(f)
+
+    def remove_channels(self, channels):
+        current_channels = self.channels
+        for channel in channels:
+            if channel in current_channels:
+                current_channels.remove(channel)
+        new_channels = ','.join(current_channels)
+        new_config = self.config
+        new_config.set(self.INI_SECTION, self.INI_OPTION, new_channels)
+        with open(self.ini_file_path, 'wb') as f:
+            new_config.write(f)
+
+
+class UpdateChannel(object):
+    """Class to handle the update channel as listed in channel-prefs.js
+    """
+
+    REGEX_UPDATE_CHANNEL_PREF = re.compile(r'("app\.update\.channel", ")([^"].*)(?=")')
+
+    def __init__(self, software_update):
+
+        self.marionette = software_update.marionette
+        self.prefs = software_update.prefs
+
+        self.config_file_path = self.marionette.execute_script("""
+            Cu.import('resource://gre/modules/Services.jsm');
+
+            var file = Services.dirsvc.get('PrfDef', Ci.nsIFile);
+            file.append('channel-prefs.js');
+
+            return file.path;
+
+            """)
+
+    @property
+    def config_file_contents(self):
+        with open(self.config_file_path) as f:
+            return f.read()
+
+    @property
+    def channel(self):
+        return self.prefs.get_pref('app.update.channel', True)
+
+    @property
+    def default_channel(self):
+        """Get the default update channel
+
+        :returns: Current default update channel
+        """
+        matches_list = re.findall(self.REGEX_UPDATE_CHANNEL_PREF, self.config_file_contents)
```

Comment:
We only have a single entry here. What would be the benefit of `findall` compared to `search`?